README.md

```diff
@@ -2,4 +2,120 @@
 
 # securedrop-export
 
-code for exporting from the securedrop qubes workstation
+Code for exporting and printing files from the SecureDrop Qubes Workstation.
+
+
+## Export Archive Format
+
+Export archive format is defined as a gzipped tar archive whose extension ends with .sd-export.
+
+### Archive Contents
+
+Once extracted, the archive will contain two elements:
+
+`metadata.json` : file containing export metadata, a file containing information about the archive and the export operation
+
+`export_data`: folder containing the raw files to export
+
+Example archive structure:
+
+```
+.
+├── metadata.json
+└── export_data
+    ├── file-to-export-1.txt
+    ├── file-to-export-2.pdf
+    ├── file-to-export-3.doc
+    [...]
+```
+
+### Archive Metadata
+
+Metadata contains three possible keys which may contain several possible values:
+`device`
+`device` specifies the method used for export, and can be either a device or a preflight check. See the Devices section below for possible values. It is a required key.
+
+`encryption_method`:
+`encryption_method` is used exclusively when exporting to USB storage. It is an optional key. Possible values are:
+luks
+
+`encryption_passphrase`
+`encryption_passphrase` is used exclusively when exporting to USB storage. It is an optional key. It contains an arbitrary string that contains the disk encryption passphrase of the device.
+
+
+Example archive metadata (`metadata.json`):
+```
+{
+  "device": "disk"
+  "encryption-method": "luks"
+  "encryption-key": "Your encryption passphrase goes here"
+}
+```
+
+### Devices
+
+For all 5 devices described below, there are three generic errors that apply:
```

Comment:
One more question, @emkll. The passage here states that "there are three generic errors that apply" but the succeeding list is five elements long. If I understand correctly, all five "generic" messages can be returned by _any_ of the device type integrations (enumerated below). 

If that's the case, perhaps we should say "For all device types (described in detail below), the following standard error types can be returned:". Then, below, instead of "The list of devices are as follows" should be changed to "The supported device types for export are as follows, including the possible errors specific to that device type:". 

What do you think?